python/semantic_kernel/kernel.py

```diff
@@ -235,47 +239,82 @@ def _create_semantic_function(
-                    self._log,
+            function.set_chat_configuration(
+                ChatRequestSettings.from_completion_config(
+                    function_config.prompt_template_config.completion
                 )
             )
-        elif backend.backend_type == BackendType.OpenAI:
-            Verify.not_null(backend.open_ai, "OpenAI configuration is missing")
-            function.set_ai_backend(
-                lambda: OpenAITextCompletion(
-                    backend.open_ai.model_id,  # type: ignore
-                    backend.open_ai.api_key,  # type: ignore
-                    backend.open_ai.org_id,  # type: ignore
-                    self._log,
+
+            if backend.backend_type == BackendType.AzureOpenAI:
+                # TODO: azure impl
+                pass
+            elif backend.backend_type == BackendType.OpenAI:
+                Verify.not_null(backend.open_ai, "OpenAI configuration is missing")
+                function.set_chat_backend(
+                    lambda: OpenAIChatCompletion(
+                        backend.open_ai.model_id,  # type: ignore
+                        backend.open_ai.api_key,  # type: ignore
+                        backend.open_ai.org_id,  # type: ignore
+                        self._log,
+                    )
                 )
-            )
+            else:
+                raise AIException(
+                    AIException.ErrorCodes.InvalidConfiguration,
+                    f"Unknown/unsupported backend type: {backend.backend_type.name}, "
+                    f"unable to prepare semantic function. Function description: "
+                    f"{function_config.prompt_template_config.description}",
+                )
+
         else:
-            raise AIException(
-                AIException.ErrorCodes.InvalidConfiguration,
-                f"Unknown/unsupported backend type: {backend.backend_type.name}, "
-                f"unable to prepare semantic function. Function description: "
-                f"{function_config.prompt_template_config.description}",
+            backend = self._config.get_completion_backend(
+                function_config.prompt_template_config.default_backends[0]
+                if len(function_config.prompt_template_config.default_backends) > 0
+                else None
             )
 
+            function.set_ai_configuration(
+                CompleteRequestSettings.from_completion_config(
+                    function_config.prompt_template_config.completion
+                )
+            )
+
+            if backend.backend_type == BackendType.AzureOpenAI:
+                Verify.not_null(
+                    backend.azure_open_ai, "Azure OpenAI configuration is missing"
+                )
+                function.set_ai_backend(
+                    lambda: AzureTextCompletion(
+                        backend.azure_open_ai.deployment_name,  # type: ignore
```

Comment:
Why are these # type: ignore comments needed here? What was the error you were getting without them?

Reply:
This is annoying, but in the `lambda: ` we get the type checker warning about `backend.azure_open_ai` possibly being `None`. This shouldn't be the case, but it'd be clumsy to try and add an assert before the lambda... really, the lambda is unnecessary here (the C# impl wants to allow for "lazy" backend creation, which is fine, but in this instance we don't need that).

There are a few ways to fix this, let's address in a follow-on PR (especially after we chat more about configuration/indirection etc.)